rclpy/rclpy/parameter.py

```diff
@@ -211,3 +260,81 @@ def parameter_value_to_python(parameter_value: ParameterValue):
         raise RuntimeError(f'unexpected parameter type {parameter_value.type}')
 
     return value
+
+
+def parameter_dict_from_yaml_file(
+    parameter_file: str,
+    use_wildcard: bool = False,
+    target_nodes: Optional[List[str]] = None,
+    namespace: str = ''
+) -> Dict[str, ParameterMsg]:
+    """
+    Build a dict of parameters from a YAML file.
+
+    Will load all parameters if ``target_nodes`` is None or empty.
+
+    :raises RuntimeError: if a target node is not in the file
+    :raises RuntimeError: if the file is not in the same format as provided by ros2 param dump
```

Comment:
Again, this seems like a circular definition. I would refrain from depending on any documentation or behavior of downstream packages, especially if that dependency is relying on the implementation here.